Merge back vNext thaliMobileNative for ios.

tobybrad opened the pull request:
<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/501)
<!-- Reviewable:end -->


<!---
@huboard:{"order":0.04914665222167969,"milestone_order":501.0,"custom_state":"ready"}
-->

ThaliTester commented:
PR is added to the queue for testing as 1. task. (f206881)

ThaliTester commented:
Test (Fail) 57531243 build is completed (f206881)

See https://github.com/ThaliTester/TestResults/tree/57531243f206881_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (c766d4e)

ThaliTester commented:
PR is added to the queue for testing as 2. task. (0087a60)

ThaliTester commented:
Test (Success) 57531243 build is completed (0087a60)

See https://github.com/ThaliTester/TestResults/tree/575312430087a60_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 57531243c766d4e(c766d4e) has failed

See https://github.com/ThaliTester/TestResults/tree/57531243c766d4e_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

ThaliTester commented:
Test 575312430087a60(0087a60) has failed

See https://github.com/ThaliTester/TestResults/tree/575312430087a60_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (acc79b5)

ThaliTester commented:
Test (Fail) 57531243 build is completed (acc79b5)

See https://github.com/ThaliTester/TestResults/tree/57531243acc79b5_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (980d818)

ThaliTester commented:
Test (Fail) 57531243 build is completed (980d818)

See https://github.com/ThaliTester/TestResults/tree/57531243980d818_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (5db8e90)

ThaliTester commented:
Test (Success) 57531243 build is completed (5db8e90)

See https://github.com/ThaliTester/TestResults/tree/575312435db8e90_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 575312435db8e90(5db8e90) has failed

See https://github.com/ThaliTester/TestResults/tree/575312435db8e90_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (1be71d2)

ThaliTester commented:
Test (Success) 57531243 build is completed (1be71d2)

See https://github.com/ThaliTester/TestResults/tree/575312431be71d2_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 575312431be71d2(1be71d2) has failed

See https://github.com/ThaliTester/TestResults/tree/575312431be71d2_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (05d96c2)

ThaliTester commented:
Test (Success) 57531243 build is completed (05d96c2)

See https://github.com/ThaliTester/TestResults/tree/5753124305d96c2_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 5753124305d96c2(05d96c2) has failed

See https://github.com/ThaliTester/TestResults/tree/5753124305d96c2_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (8097721)

ThaliTester commented:
Test (Success) 57531243 build is completed (8097721)

See https://github.com/ThaliTester/TestResults/tree/575312438097721_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 575312438097721(8097721) has failed

See https://github.com/ThaliTester/TestResults/tree/575312438097721_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (1f256a6)

ThaliTester commented:
Test (Success) 57531243 build is completed (1f256a6)

See https://github.com/ThaliTester/TestResults/tree/575312431f256a6_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 575312431f256a6(1f256a6) has failed

See https://github.com/ThaliTester/TestResults/tree/575312431f256a6_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

yaronyg commented:
Reviewed 48 of 64 files at r1, 9 of 13 files at r3, 1 of 1 files at r4, 3 of 5 files at r5, 1 of 1 files at r6, 1 of 2 files at r7, 1 of 1 files at r8, 1 of 2 files at r9, 1 of 1 files at r10, 1 of 1 files at r11.
Review status: all files reviewed at latest revision, 30 unresolved discussions.

---

*[src/ios/JXcoreExtension.m, line 56 \[r4\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Ev6DBq2YD7ixWlX6O:-K9Ev6DCclDyv7OFMSPW:1219978150) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/0087a600917c35a009aaae7751f792c11327e0ef/src/ios/JXcoreExtension.m#L56)):*
I thought JXcore now had a way to fire events without having to be on the main thread?

---

*[src/ios/JXcoreExtension.m, line 114 \[r4\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9EzuVjiQXlgoe-BYnT:-K9EzuVjiQXlgoe-BYnU:-825361631) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/0087a600917c35a009aaae7751f792c11327e0ef/src/ios/JXcoreExtension.m#L114)):*
Please help educate me here. What happens if this exception gets thrown? Does Thali just crash?

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.h, line 65 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YSb3zQFCF2M1B8fR5:-K9YSb3zQFCF2M1B8fR6:831473031) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.h#L65)):*
/connectio/connection

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.m, line 62 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YVHctjpaDHdzq3VcX:-K9YVHcuufH6ZyvYvdS2:1252528537) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.m#L62)):*
How do we keep this class from getting so full we run out of RAM? More generally the spec calls for limiters to make sure that if we get overwhelmed with traffic we shut down rather than trying to handle it. I didn't see those limiters. If they aren't being checked in here can you please open a bug for when they are supposed to be implemented?

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.m, line 161 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YTGX2ezwcbR27E5OH:-K9YTGX3_jNN0VukhESb:-1887782018) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.m#L161)):*
In the parent "if" we checked that serverSession connectionState != THEPeerSessionStateConnected and we can only reach this code because that was true. But then here we check that they are equal, which we just checked wasn't true! Unless you are checking for some kind of multi-threaded race condition how could this if statement ever be true? One wonders if the second part of the and in the enclosing IF shouldn't be there?

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.m, line 361 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YWmyQ7lDMUPtDNGJH:-K9YWmyQ7lDMUPtDNGJI:1547823789) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.m#L361)):*
Shouldn't we at least make sure there is nothing sleezy going on here? For example, shouldn't we validate that peerUUID doesn't equal our own UUID? I'm just paranoid because this is some random string taken from the air.

---

*[src/ios/MultipeerConnectivity/THEMultipeerClientSession.m, line 73 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YXdrOo_ee-rWcdlxb:-K9YXdrOo_ee-rWcdlxc:542097187) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClientSession.m#L73)):*
/callabck/callback

---

*[src/ios/MultipeerConnectivity/THEMultipeerManager.m, line 167 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YvonW7jl74jDofIUa:-K9YvonW7jl74jDofIUb:-1341210942) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerManager.m#L167)):*
Please humor me and please put in brackets

---

*[src/ios/MultipeerConnectivity/THEMultipeerManager.m, line 292 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YwDRfsLlE-PnU9ojg:-K9YwDRgdcZDwNN1anPA:-1607348736) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerManager.m#L292)):*
If we don't need this why not just remove it? I'm guessing this is a left over from when listening and advertising could be done separately?

---

*[src/ios/MultipeerConnectivity/THEMultipeerManager.m, line 366 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YwiiLhqe5iEFp3MK8:-K9YwiiLhqe5iEFp3MK9:774703514) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerManager.m#L366)):*
Why is this disabled?

---

*[src/ios/MultipeerConnectivity/THEMultipeerServer.m, line 176 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YQQtWnSpEEuLG7qaR:-K9YQQtWnSpEEuLG7qaS:-499859813) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerServer.m#L176)):*
/constiuent/constituent

---

*[src/ios/MultipeerConnectivity/THEMultipeerServer.m, line 177 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YQrMowWdhcV1oDJiM:-K9YQrMp52mFqXyvWNgk:-973238064) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerServer.m#L177)):*
This call makes me nervous because if I'm following the code correctly this data is being passed into us directly by iOS who got it directly from the other caller. Do we know what, if any, work iOS does to make sure that whatever is in context can't hurt us? Do we need to check if it is null? If it can be in a format that could cause some problem with alloc? If it has a reasonable length?

---

*[src/ios/MultipeerConnectivity/THEMultipeerServer.m, line 186 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YRaLbDNALVVoqwhda:-K9YRAoq1OpZw4jbCDMN:-51646146) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerServer.m#L186)):*
What happens if these are malformed? Remember, we are taking arbitrary data off the net. Danger Will Robinson? Don't we need to check?

---

*[src/ios/MultipeerConnectivity/THEMultipeerServer.m, line 223 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YSEpLtbLG3jln3hdq:-K9YSEpMo9ip9e28u6c-:1998043204) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerServer.m#L223)):*
Why not stick in a return and get rid of the else?

---

*[src/ios/MultipeerConnectivity/THEMultipeerSocketRelay.m, line 273 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YyuEMokrXzhFHBcD0:-K9YyuEMokrXzhFHBcD1:1333817467) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerSocketRelay.m#L273)):*
I'm curious why you changed from a const to a define?

---

*[src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m, line 130 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z01UhjDgsH268DMiM:-K9Z01UhjDgsH268DMiN:-1005973123) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m#L130)):*
I think the second app2 in the sentence above is meant to be app1

---

*[src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m, line 315 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z0sh_e2AcivCIQU8a:-K9Z0sh_e2AcivCIQU8b:-1455535077) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m#L315)):*
But shouldn't this test fail? In other words if a peer sends an invite to UUID Foo and then the server changes its UUID to be Bar then shouldn't an old request to Foo fail?

---

*[src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m, line 498 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z1Ftly8wX_b__RyxC:-K9Z1Ftly8wX_b__RyxD:-250023793) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m#L498)):*
Wouldn't a large bit of data perhaps test things better? Say at least a meg or two?

---

*[src/ios/proj/Thali/TestAppTests/ThaliSingleInstanceTests.m, line 41 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z1bdkh4uPzoZnPEp1:-K9Z1bdkh4uPzoZnPEp2:539693801) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/proj/Thali/TestAppTests/ThaliSingleInstanceTests.m#L41)):*
It would be great to also test that we can call stop several times in a row without an error.

Also it would probably be a good idea to have tests where we call start and stop interleaved with each other just to make sure that state restarts appropriately.

---

*[src/ios/THEAppContext.m, line 122 \[r4\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9F1JC5oqOOxlVSBQjy:-K9F1JC5oqOOxlVSBQjz:-1308394372) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/0087a600917c35a009aaae7751f792c11327e0ef/src/ios/THEAppContext.m#L122)):*
Isn't it time to remove this? The tests are all dead, right?

---

*[src/ios/THEAppContext.m, line 108 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z2BuRqBBxWBREw5_A:-K9Z2BuSlVjRrO0WcnbE:-1390298949) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/THEAppContext.m#L108)):*
Is there a 10 foot tall warning somewhere that both startUpdateAdvertising and startListening have to both be on or connections won't work?

---

*[src/ios/THEAppContext.m, line 169 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z2RsBgaeNqdH9UR4O:-K9Z2RsBgaeNqdH9UR4P:-79262649) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/THEAppContext.m#L169)):*
Same comment as above, time to retire this?

---

*[src/ios/THEAppContext.m, line 222 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z2oCUfAUl9CO2Rjw8:-K9Z2oCV3nUsCZ5oTpfC:923737292) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/THEAppContext.m#L222)):*
Sigh... I hate having this code still here. We aren't using it and it isn't clear when we will use it. The problem is that it's a major goal to eventually use this to enable the BLE hub functionality. I guess we just have to leave it here? We could at least mark it as not being called?

---

*[src/ios/THEAppContext.m, line 341 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z3LhejPDiY2xz8Z9j:-K9Z3LhfiPv9IbaD-Cey:-1635246261) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/THEAppContext.m#L341)):*
At least according to the spec these aren't the right values for network changed. See [here](https://github.com/thaliproject/Thali_CordovaPlugin/blob/vNext/thali/NextGeneration/thaliMobileNative.js#L435)

---

*[src/ios/THEThaliEventDelegate.h, line 37 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z3j-UPxmzNHNNklz0:-K9Z3j-UPxmzNHNNklz1:860672446) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/THEThaliEventDelegate.h#L37)):*
Are we just not implementing discoveryAdvertisingStateUpdate? That's probably o.k. We should wait until somebody needs it but I wanted to make sure this was intentional and not just an accident?

What about incomingConnectionToPortNumberFailed? That seems pretty important.

---

*[test/TestServer/UnitTestConfig.js, line 13 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z4Qvg8Ml08y2crqGi:-K9Z4QvhZpGtq6grYEBG:-1033318960) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/TestServer/UnitTestConfig.js#L13)):*
You're checking in android set to 0?

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 67 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZFuv-2O6otHwUT5KG:-K9ZFuv0xKo5Yzq3S4zp:-1243357923) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L67)):*
As I mentioned in the native code it would be really good to have start-stop-start tests just to make sure we clean up our state properly. But I think this would require more than just checking the bool that comes back. It would probably be something to stick on a test that moves data around. E.g. take that data moving tests and basically run it twice in a row.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 111 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZTbQov-YXLwZ-DGWy:-K9ZTbQpsj0nLfsctwCx:782541474) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L111)):*
Don't we also need tests for discoveryAdvertisingStateUpdateNonTCP (or a clear statement that we just aren't going to implement it for now which is fine) , networkChanged and incomingConnectionToPortNumberFailed? It's fine not to implement all of those now. But please open a new bug so we don't forget.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 112 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZTwNYkLMPVZlXuljG:-K9ZTwNYkLMPVZlXuljH:1038250002) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L112)):*
Do you intend to do a separate test for actually moving (large) amounts of data? This should be separate from the perf tests. But we should always have a good test that we can move around a few megs without issue.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 124 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZH-UPZIsUJy77id-H:-K9ZH-UPZIsUJy77id-I:92714492) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L124)):*
Don't we need to make sure these are peers we are expecting so we don't end up making a mess talking to broken clients who just happen to be around?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 2. task. (01473a8)

ThaliTester commented:
Test (Fail) 57531243 build is completed (01473a8)

See https://github.com/ThaliTester/TestResults/tree/5753124301473a8_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (fe78772)

ThaliTester commented:
Test (Fail) 57531243 build is completed (fe78772)

See https://github.com/ThaliTester/TestResults/tree/57531243fe78772_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (1745da8)

ThaliTester commented:
Test (Success) 57531243 build is completed (1745da8)

See https://github.com/ThaliTester/TestResults/tree/575312431745da8_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 575312431745da8(1745da8) has failed

See https://github.com/ThaliTester/TestResults/tree/575312431745da8_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (f4b2975)

ThaliTester commented:
Test (Fail) 57531243 build is completed (f4b2975)

See https://github.com/ThaliTester/TestResults/tree/57531243f4b2975_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (fa91ca3)

tobybrad commented:
Review status: 0 of 71 files reviewed at latest revision, 30 unresolved discussions.

---

*[src/ios/JXcoreExtension.m, line 56 \[r4\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Ev6DBq2YD7ixWlX6O:-K9g_NjqfuELry6mw3m1:-1875613254) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/0087a600917c35a009aaae7751f792c11327e0ef/src/ios/JXcoreExtension.m#L56)):*
If it does it's not obvious: https://github.com/jxcore/jxcore-ios-sample/blob/master/JXcoreSample/JXcore/JXcore.h

---

*[src/ios/JXcoreExtension.m, line 114 \[r4\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9EzuVjiQXlgoe-BYnT:-K9g_ZbblsY_0-7KtorA:1723652105) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/0087a600917c35a009aaae7751f792c11327e0ef/src/ios/JXcoreExtension.m#L114)):*
Yes. We're running as a plugin remember.. so we're really just throwing to the cordova wrapper and invoking whatever behaviour it has for uncaught exceptions. In practice this will never throw though.. unless something's trashed memory perhaps.. we're completely in control of the objects being passed here.

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.h, line 65 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YSb3zQFCF2M1B8fR5:-K9gae4mzGTGnliDsAS2:-102228352) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.h#L65)):*
/on/one

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.m, line 62 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YVHctjpaDHdzq3VcX:-K9geLcrg-W0NPrC89Zd:1383406976) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.m#L62)):*
Added a limit.. 64.. perhaps too high.

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.m, line 161 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YTGX2ezwcbR27E5OH:-K9gf9qMOISX_esMjgvP:-789615350) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.m#L161)):*
Should be checking for NotConnected. Fixed.

---

*[src/ios/MultipeerConnectivity/THEMultipeerClient.m, line 361 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YWmyQ7lDMUPtDNGJH:-K9gfMFlOeDFt7F7jYG6:354751531) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerClient.m#L361)):*
In normal operation you never discover yourself. If someone's deliberately cloning your ephemeral id then so what. Node layer can't discriminate and so you're just in the same case as them cloning someone else's id.. higher level's must take of determining an impersonation.

---

*[src/ios/MultipeerConnectivity/THEMultipeerManager.m, line 167 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YvonW7jl74jDofIUa:-K9ghk-IIJ_jkiLAFF35:-707152333) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerManager.m#L167)):*
/brackets/braces
/humor/humour

---

*[src/ios/MultipeerConnectivity/THEMultipeerManager.m, line 292 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YwDRfsLlE-PnU9ojg:-K9gi6WMw2kprOqeibSy:-2020186758) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerManager.m#L292)):*
No.. experimenting whether we need to restart both server and client to keep things ticking over.. it currently looks like just the client needs restarting to make things work. Will remove once I'm happy that's definitely the case.

---

*[src/ios/MultipeerConnectivity/THEMultipeerSocketRelay.m, line 273 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9YyuEMokrXzhFHBcD0:-K9gs5SJahKswI_5EBRb:820401264) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/MultipeerConnectivity/THEMultipeerSocketRelay.m#L273)):*
So am I.

---

*[src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m, line 315 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z0sh_e2AcivCIQU8a:-K9gtTxhiGe_M_fXT99H:-843042638) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m#L315)):*
It's not connecting to the old id.. it's connecting to the new one. Order of execution is not straight down the page. I'll put in the converse test to illustrate.

---

*[src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m, line 498 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z1Ftly8wX_b__RyxC:-K9gxOoK5hx6Pqv9T-3D:-537467337) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/proj/Thali/TestAppTests/ThaliMultipleInstanceTests.m#L498)):*
That's for a different test.

---

*[src/ios/THEAppContext.m, line 108 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z2BuRqBBxWBREw5_A:-K9h4idpehRRehBIfo7p:-1742852729) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/src/ios/THEAppContext.m#L108)):*
So.. perhaps bad.. but when users calls startAdvertising.. I start the client anyway which is what isListening is all about. Did you manually start the client or did I start one to be able to initiate reverse connection. All works as expected at the node layer.

---

*[test/TestServer/UnitTestConfig.js, line 13 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9Z4Qvg8Ml08y2crqGi:-K9h40DZZoarCyxyLh1r:1347764746) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/TestServer/UnitTestConfig.js#L13)):*
Nope.. I'm testing CI with android set to 0

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 111 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZTbQov-YXLwZ-DGWy:-K9h4FW4j-EF9a9Cr_XB:1438650066) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L111)):*
networkChanged presents challenges for testing.

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 124 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZH-UPZIsUJy77id-H:-K9h4WH0-MWsJOl32IBy:-663205177) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L124)):*
Does it really matter ? As long as we can make a connection to a peer we're good, right ?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
Test (Fail) 57531243 build is completed (fa91ca3)

See https://github.com/ThaliTester/TestResults/tree/57531243fa91ca3_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (74916c2)

ThaliTester commented:
Test (Success) 57531243 build is completed (74916c2)

See https://github.com/ThaliTester/TestResults/tree/5753124374916c2_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
Test 5753124374916c2(74916c2) has failed

See https://github.com/ThaliTester/TestResults/tree/5753124374916c2_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the fail logs

yaronyg commented:
Reviewed 53 of 76 files at r17, 18 of 19 files at r18, 3 of 3 files at r19, 2 of 2 files at r20.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[readme.md, line 43 \[r20\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9mAPTtl0YkBmjNl3d4:-K9mAPTtl0YkBmjNl3d5:912422930) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/74916c25f02d29d0c392e023fd4de512b7dca3c6/readme.md#L43)):*
I think this has now been fixed in the very latest release of JXcore.

---

*[src/ios/THEAppContext.m, line 379 \[r20\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9mHfowy59nEqdofo4D:-K9mHfowy59nEqdofo4E:1424642395) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/74916c25f02d29d0c392e023fd4de512b7dca3c6/src/ios/THEAppContext.m#L379)):*
What if the radios are turned off?

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 111 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZTbQov-YXLwZ-DGWy:-K9mWLQvdX_qCrcJFQl_:-1413033834) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L111)):*
Oh please. Just root each phone and use private APIs to turn the network on and off. Oh and while you're doing that also put in an automatic unrooter so the rest of the tests can run in a proper environment. Seriously... devs these days. :)

---

*[test/www/jxcore/bv_tests/testThaliMobileNative.js, line 124 \[r11\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9ZH-UPZIsUJy77id-H:-K9mWXwNQsPBFXxNOadk:-1181207189) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/1f256a65f8b9c16d9ab2380a938fee76731a1373/test/www/jxcore/bv_tests/testThaliMobileNative.js#L124)):*
I'm going to say yes but wow do I bet this one will bite us eventually.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 1. task. (35d934c)

ThaliTester commented:
Test (Fail) 57531243 build is completed (35d934c)

See https://github.com/ThaliTester/TestResults/tree/5753124335d934c_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (4672864)

ThaliTester commented:
Test (Fail) 57531243 build is completed (4672864)

See https://github.com/ThaliTester/TestResults/tree/575312434672864_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (33467de)

ThaliTester commented:
Test (Fail) 57531243 build is completed (33467de)

See https://github.com/ThaliTester/TestResults/tree/5753124333467de_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (afeb1f4)

ThaliTester commented:
Test (Fail) 57531243 build is completed (afeb1f4)

See https://github.com/ThaliTester/TestResults/tree/57531243afeb1f4_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (aaa3517)

ThaliTester commented:
Test (Fail) 57531243 build is completed (aaa3517)

See https://github.com/ThaliTester/TestResults/tree/57531243aaa3517_Merge_back_vNext_thaliMobileNative_for_ios__tobybrad/ for the logs

msftclas commented:
__@tobybrad__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;

tobybrad commented:
Review status: 59 of 69 files reviewed at latest revision, 11 unresolved discussions.

---

*[readme.md, line 43 \[r20\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501#-K9mAPTtl0YkBmjNl3d4:-KAF_HvgO1WbqCsktuxk:-499146009) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/74916c25f02d29d0c392e023fd4de512b7dca3c6/readme.md#L43)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/501)*
<!-- Sent from Reviewable.io -->
